### PR TITLE
Simple robots.txt for server 404 log

### PR DESCRIPTION
**Pull request by Shnoulle**
- See https://github.com/friendica/friendica/issues/3334
- Most simple possible, unsure for other needed disallowed part for all ?
- Don't find the inherited doc for `****_init`

### CODE REVIEW

**Comment by Shnoulle** [comment]
@Hypolite can do more but with more times to see what i can get from config to disable other part (some user can set more private part for example). But finally : friendica didn't give public access to 'private' part , then not really an issue.

**Comment by Hypolite** [comment]
Take your time, this isn't a critical task.

I was wondering why you had the following leading block with an empty `Disallow` list before the actual one?

Also I would suggest you to use the `robots_txt_content()` magic function, it seems more appropriate to me since you're outputting something.

### Standards

- Please use tabs for indentation
- Please wrap operators (`=`, `==`, `=>`, `+`, `-`, etc...) with spaces
- Please add a space after control structure names like `if`, `foreach`, `switch`
- Please use a single string quoting style, preferably single-quotes
- Please use `PHP_EOL` constant instead of literal `"\n"`

### Documentation

I don't believe there is an inherited doc for `{module}_init()`, `{module}_content()` and `{module}_post()` functions, but I'll let @tobiasd prove me wrong.

**Comment by fabrixxm** [comment]
As  `*_init()`, `*_content()` etc are global functions and not class methods, I believe that there aren't docs to inherit...

**Comment by Shnoulle** [comment]
1. About `PHP_EOL` : unsure because robots.txt are display on any browser .
2. About tab : Ah ? Do you have a link to conding standard of friendica please ? 
3. What is `robots_txt_content()` magic function ?
4. empty Disallow ... oups
4. About inherit : yes , not inherit. but a lot of _init have exactly the same then :/

```
 * @param App $a
 * @return void
```

**Comment by Shnoulle** [comment]
About `***_content` : sorry, didn't understand ....

1. It's not really a content of the web site : then init seems OK IMHO
2. Still have header , then need die/quit

File: mod/robots_txt.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * return the default robots.txt
```
Comment (rabuzarus) [inline mod/robots_txt.php]: please use standard doxygen syntax. For file docu
```
/**
 * @file mod/robots_text.php
 * @brief Module which returns the default robots.txt
 * @version 0.1.0
 */
```

**Comment by Hypolite** [comment]
About `PHP_EOL`: It doesn't matter for browsers, all of them recognize all the different newline style, as a classic `robots.txt` file would feature the newline style of the underlying OS of the web server anyway.

About the coding standards, there are all there : https://social.shnoulle.net/help/Developers-Intro#1_5_2

About the `{module}_content` function, it is shortly described here: https://github.com/friendica/friendica/blob/develop/index.php#L223

You may be right about usage of `{module}_init()` though, if it loads the whole page wrapper.

File: mod/robots_txt.php [hunk 2]
```diff
@@ -1,29 +1,27 @@
 <?php
 /**
  * return the default robots.txt
- * @version 0.1.0
+ * @version 0.1.1
  */
 
 /**
  * Simple robots.txt
```
Comment (rabuzarus) [inline mod/robots_txt.php]: STANDARDS: please use ` * @brief Simple robots.text`

File: mod/robots_txt.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+/**
+ * Simple robots.txt
+ * @param App $a
+ * @return void
+ */
+function robots_txt_init(App $a) {
+
+	/** @var string[] globally disallowed url */
```
Comment (rabuzarus) [inline mod/robots_txt.php]: please use `/*` or `//` for comments. If you youse `/**` it would be added to the doxygen documentation

**Comment by Shnoulle** [comment]
Sorry , using [phpdoc](https://www.phpdoc.org/docs/latest/references/phpdoc/tags/var.html) , not doxygen here . I'm a bad boy, didn't read [the manual](https://social.shnoulle.net/help/Developers-Intro)

File: mod/robots_txt.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * @brief Return default robots.txt when init
+ * @param App $a
+ * @return void
+ */
+function robots_txt_init(App $a)
+{
+
+	/** @var string[] globally disallowed url */
```
Comment (rabuzarus) [inline mod/robots_txt.php]: Just on last little thing. Is this also phpdoc style? All comments with `/**` at the beginning will be inserted into the doxygen docu.

Comment (Shnoulle) [inline mod/robots_txt.php]: Yep : https://www.phpdoc.org/docs/latest/references/phpdoc/tags/var.html#examples . I'm a lazy one , never use tool , just geany ;)

**Comment by tobiasd** [comment]
On Wed, 26 Apr 2017 07:44:51 -0700
Hypolite Petovan <notifications@github.com> wrote:

> I don't believe there is an inherited doc for `{module}_init()`, `
> {module}_content()` and `{module}_post()` functions, but I'll let
> @tobiasd prove me wrong.

I think this is one part where the code is the documentation ;-)

**Comment by Hypolite** [comment]
@tobiasd Nah, the *_content() function loads the whole site wrapper, which we need for `/friendica` but not for `robots.txt`. Look at `mod/_well-known.php` for example.

**Comment by tobiasd** [comment]
Yes I see @Hypolite, just had to look a bit more into the code.

But, the `$allDisalloweds` variable should be something that is set in the `.htconfig.php` file (or the admin panel which can be added later) and look something like `$a->config[''system']['robots_disallow']` or so. As it is configurable or should this only be the fall-back solution for lazy admins who don't care about the robots.txt?

**Comment by Shnoulle** [comment]
@tobiasd : didn't know if it's for lazy admin or not, I can get it in `$a->config['system']['robots_disallow']` starting without GUI. 

But about _content, maybe need another magic function where content is not really content ? And looking at code : init have $a if needed too.

**Comment by Hypolite** [comment]
As it is supposed to be a default robots.txt handler, I don't think there's a need for a config variable. For any other robots.txt needs a standalone file can be created.

**Comment by tobiasd** [comment]
@rabuzarus are you satisfied with the in code documentation?

**Comment by rabuzarus** [comment]
I'm a little bit unsure about `/** @var string[] globally disallowed url */` I would say we don't need it as a doxygen docu.

**Comment by Shnoulle** [comment]
Like you want, currently : think all this part is not really needed, just to directly disallow respectfull bot to get a 401. Then can be directly removed.

**Comment by Hypolite** [comment]
I think he was only referring to the documentation comment, not the the actual implementation.

**Comment by Shnoulle** [comment]
I know :) . And we can remove the 2 part : and the implementation and the comment ;)

**Pull request merged**